tagmanager/src/tm_ctags_wrappers.h

```diff
@@ -0,0 +1,49 @@
+/*
+ *      tm_ctags_wrappers.h - this file is part of Geany, a fast and lightweight IDE
+ *
+ *      Copyright 2016 Jiri Techet <techet(at)gmail(dot)com>
+ *
+ *      This program is free software; you can redistribute it and/or modify
+ *      it under the terms of the GNU General Public License as published by
+ *      the Free Software Foundation; either version 2 of the License, or
+ *      (at your option) any later version.
+ *
+ *      This program is distributed in the hope that it will be useful,
+ *      but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *      MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *      GNU General Public License for more details.
+ *
+ *      You should have received a copy of the GNU General Public License along
+ *      with this program; if not, write to the Free Software Foundation, Inc.,
+ *      51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef TM_CTAGS_WRAPPERS
+#define TM_CTAGS_WRAPPERS
+
+#include <glib.h>
+
+#include "tm_parser.h"
+
+#include "entry.h" /* for sTagEntryInfo */
+
+
+G_BEGIN_DECLS
+
+typedef gboolean (*tm_ctags_callback) (const tagEntryInfo *const tag,
+	gboolean invalidate, void *user_data);
```

Comment:
It also seems to have a strange signature only targeted at our very specific usage, although it suggest it's meant to be a generic API.  By that I mean the `invalidate` parameter, that seem more like a step indicator that might be better served as an enum, or at least rather name it something like "initialize" or something more generic.

But maybe
```C
typedef enum TMCtagsParsingStep
{
	TM_CTAGS_PARSING_STEP_INITIALIZE,
	TM_CTAGS_PARSING_STEP_EMIT_TAG,
	// ... possibly _FINALIZE too
}
```
Or something. Or, if we can be sure `tag` is never `NULL` when emitting tags, rely only on this?

Also, documentation on this callback would be great :)